### ISSUE TITLE
Extra help for tasks

### ISSUE THREAD

**Issue by jwoertink** [luckyframework/lucky#786]
When you run `lucky -h`, you get output like:

```
Usage: lucky name.of.task [options]

Available tasks:

  ▸ build.release             Compile app for production
  ▸ db.create                 Create the database
  ▸ db.create_required_seeds  Add database records required for the app to work
  ▸ db.create_sample_seeds    Add sample database records helpful for development
  ▸ db.drop                   Drop the database
  ▸ db.migrate                Migrate the database
  ▸ db.migrate.one            Run the next pending migration
  ▸ db.redo                   Rollback then run the last migration
  ▸ db.rollback               Rollback the last migration
  ▸ db.rollback_all           Rollback all migrations
  ▸ exec                      Execute code. Use this in place of a console/REPL
  ▸ gen.action.api            Generate a new api action
  ▸ gen.action.browser        Generate a new browser action
  ▸ gen.component             Generate a new component
  ▸ gen.migration             Generate a new migration
  ▸ gen.model                 Generate a model, query, and form
  ▸ gen.page                  Generate a new page
  ▸ gen.resource.browser      Generate a resource (model, form, query, actions, and pages)
  ▸ gen.secret_key            Generate a new secret key
  ▸ routes                    Show all the routes for the app
  ▸ watch                     Start and recompile project when files change
```

For someone brand new to lucky, this is a little daunting, as well as slightly confusing.  You know that you can run a migration like `lucky gen.migration`, and you know it takes options like `[options]`, but you don't know **what** that actually looks like. _What_ `[options]` can you pass? 

For this case, it would be nice if you could do `lucky -h gen.migration` and then see output similar to:

```
Usage: lucky gen.migration AddColumnToModel

Example:

  ▸ lucky gen.migration AddNameToUsers     
```

Of course this would mean that CLI would need an optional `example` macro so your custom tasks, or tasks created by shards could be looped in.

**Comment by paulcsmith** [comment]
I like this idea, but I think we can rely on each task to implement it (`lucky gen.migration --help`), rather than making `lucky -h` command. 

I think it'd be nice to make it easier for commands to implement this. It could be an `example` macro or maybe a `help_message` method you could define and could return whatever you want.

```crystal
class MyTask < LuckyCli::Task
  def help_message
     "Something"
  end
end
```

And under the hood LuckyCli::Task will look for `-h`, `help` or `--help` and if it has those args then it will print the help message. Does that make sense?

**Comment by jwoertink** [comment]
Yup. I like that too. Right now you just get an unexpected result

```
[10:11AM] junk_blog (master)$ lucky gen.migration -h
Created -h::V20190531101121 in ./db/migrations/20190531101121_-h.cr
```
😂

**Comment by paulcsmith** [comment]
Haha yeah that's no good. Luckily I think this would be a fairly easy fix.

We'd have to change this line from `task.call` to `task.help_or_call` and then a new method in `LuckyCli::Task#help_or_call` that checks the args and if it is one of the help options it calls `help_message`. By default the help message can just be something like "Run this with lucky task.name <ARGS>" so that *all* tasks have a help message but can be easily overridden.

Thoughts?

**Comment by jwoertink** [comment]
I like it it. The only thing that comes to mind might be a super edge case, but running something like `lucky gen.resource.browser help` could possibly conflict with if someone is building helpdesk software where they want a resource called "help". We could also just wait for that issue to come up before addressing it.

**Comment by paulcsmith** [comment]
Though most of those tasks require a class name so that would be invalid `Help` so might be ok! I agree we can go with it and fix if needed. If it comes up we could have an overridable method called `help_args` or something. Defaults to the regular help list but can be just `-h` or `--help` if totally necessary

**Comment by paulcsmith** [comment]
LMK if you wanna tackle this. If not I'll try to find some time this weekend!

**Comment by jwoertink** [comment]
Yeah I can take that.

**Comment by paulcsmith** [comment]
I'm gonna move this to Lucky CLI
